js/__internal/grids/grid_core/views/m_grid_view.ts

```diff
@@ -57,10 +57,21 @@ const resizingControllerMembers = {
 
     if (!this._refreshSizesHandler) {
       this._refreshSizesHandler = (e) => {
+        let resizeDeferred;
```

Comment:
Let's modify this block of the code a little:
* add type to ```resizeDeferred```
* put condition (70 line) out of brackets in a separate variable
* pass to the ```when``` function resolved ```Deferred``` instead of the ```undefined```
```when``` function can handle the ```undefined``` value, but let's make it more obvious for the future refactoring :)

E.g. something like this:
```typescript
let resizeDeferred: DeferredObj<unknown> | null = null;
const changeType = e?.changeType;
const isDelayed = e?.isDelayed;
// I'm out of the context to come up with this variable name :)
// I guess that you can come up with a better name here
const suitableChangeType = changeType
  && changeType !== 'updateSelection'
  && changeType !== 'updateFocusedRow'
  && changeType !== 'pageIndex'
  && !isDelayed;

dataController.changed.remove(this._refreshSizesHandler);

if (this._checkSize()) {
  resizeDeferred = this._refreshSizes(e);
}

if (suitableChangeType) {
  resizeDeferred ??= new DeferredObj<null>().resolve(null);
  when(resizeDeferred).done(() => {
    this._setAriaLabel();
    this.fireContentReadyAction();
  });
}
```